src/client/driver/driver.js

```diff
@@ -1651,6 +1651,12 @@ export default class Driver extends serviceUtils.EventEmitter {
 
         const role = await this._getDriverRole();
 
+        // NOTE: the child window can become master during the preceding async requests
+        // in this case we do not need to call the `_startInternal` method again
+        // since it was called during the `_handleSetAsMasterMessage` method.
+        if (this.role === DriverRole.master)
```

Comment:
It should be places before the ` const role = await this._getDriverRole();` instruction.

Reply:
No. It should be placed after all async requests. Since while the `_getDriverRole` request is executing we have a chance, that the current window is marked as master inside the `_handleSetAsMasterMessage` handler. Then we call `startCommandProcessing` method inside the `_handleSetAsMasterMessage`. Then we call to `_getDriverRole` method from `init` and it returns `master`. Then we call `startInternal` and `startCommandProcessing` again from `init`. Thus we call the `startCommandProcessing` twice. It leads to error.

In my case. We check `this.role === DriverRole.master` after all async requests. If after requests `this.role === master` it means that commands were already started and we do not need to start if again.

Reply:
Ok. Thank you for the detailed clarification.